docs/rfcs/0005-json-library.md

```diff
@@ -0,0 +1,48 @@
+---
+layout: page
+title: RFC process
+permalink: /rfc/0005-json-library
+---
+
+* Start date: 2020-07-28
+* RFC PR: TBD
+
+## Summary
+
+Introduce a library for reading and writing JSON into Android Components and Fenix.
+
+## Motivation
+
+We currently interact with JSON in many areas of Android Components and Fenix. Currently serialization and deserialization is all done manually with `JSONObject`, which creates more work for developers and potential bugs. Fenix also is interested in using a library and we'd like to avoid shipping multiple JSON libraries to users. Using a JSON library allows us to reduce code that needs to be written, potentially gain some performance wins from a well maintained lib, and avoid bugs in our custom JSON implementations.
+
+Some examples of JSON usage today are:
+- Reading a `WebAppManifest`
+- Parsing API calls in various service components
+- Sending messages to and from extensions
+
+## Guide-level explanation
+
+We'll start with selecting a JSON library to integrate into Android Components and Fenix. Some options we've looked at include:
+- [kotlinx-serialization](https://github.com/Kotlin/kotlinx.serialization): Multi-platform, multi-format, built by JetBrains. Also much newer than other solutions so documentation is relatively lacking.
+- [Gson](https://github.com/google/gson): Long standing library built by Google. Slower than other options.
+- [Moshi](https://github.com/square/moshi): Simpler API, built by Square. Fewer features than Gson but also faster.
+
+Once a library is chosen we can use it when building new components that use JSON and slowly migrate existing code.
+
+## Drawbacks
+
+* Using a new tool means we have to invest some time in learning how to use it. We're at least familiar with JSONObject as are most people. This "having another thing to learn" thing is especially problematic if Fenix and other apps adopt different technologies to fill the same gap.
+* We want to be conservative adding third-party libraries.
```

Comment:
Our design axioms list why we want to be conservative, maybe we can repeat some of those arguments here as drawbacks: https://mozac.org/contributing/design-axioms, we could list here.

Specifically APK size is something that we should consider and explicitly measure for the options here before making a decision.